arcgis-js-api/arcgis-js-api.d.ts

```diff
@@ -1227,6 +1229,8 @@ declare module "esri" {
     highlightLocation?: boolean;
     /** The infoTemplate used for the highlight graphic. */
     infoTemplate?: InfoTemplate;
+    /** Goes to the users extent. */
+    locate?: () => any;  
```

Comment:
The LocateButton does have a `locate` method and event, but not a constructor option. See:
https://developers.arcgis.com/javascript/jsapi/locatebutton-amd.html